### PR TITLE
Create a scaffold for a search app.

### PR DESCRIPTION
**Pull request by jsuereth**
This just sets up an initial architecture with all components in place to make it easier for someone to learn Ratatui/Crosstool with a resolved schema and implement searching features.

The goal of this is NOT completeness. It's meant to be "bare bones" enough someone can take this and make it whatever in the future, but also NOT a burden for us to evolve weaver going forward if it exists.

Hopefully someone wanting to learn Rust and/or Weaver sees this and decided to contribute/drive this forward.

### CODE REVIEW

**Comment by lquerel** [comment]
This is the commit where I removed the search command before the release 0.1.0. That should help initiate this task.

https://github.com/open-telemetry/weaver/commit/96e2c134214c237cf3c70e57b90ed7dbd08763d1

File: src/registry/search.rs [hunk 1]
```diff
@@ -17,7 +40,96 @@ pub struct RegistrySearchArgs {
...
+    let registry_path =
+        semconv_registry_path_from(&args.registry.registry, &args.registry.registry_git_sub_dir);
+
+    // Load the semantic convention registry into a local cache.
+    let semconv_specs = load_semconv_specs(&registry_path, cache, logger.clone())?;
+    let mut registry = SemConvRegistry::from_semconv_specs(registry_id, semconv_specs);
+    let schema = resolve_semconv_specs(&mut registry, logger.clone())?;
+
+    // TODO - We should have two modes:
+    // 1. An interactive UI
```
Comment (lquerel) [inline src/registry/search.rs:132]: For the interactive mode, see this commit https://github.com/open-telemetry/weaver/commit/96e2c134214c237cf3c70e57b90ed7dbd08763d1 path=src/search.

File: Cargo.toml [hunk 2]
```diff
@@ -74,6 +74,9 @@ weaver_forge = { path = "crates/weaver_forge" }
 weaver_checker = { path = "crates/weaver_checker" }
 
 clap = { version = "4.5.4", features = ["derive"] }
+rayon = "1.10.0"
+ratatui = { version = "0.26.3", features=["serde"] }
+crossterm = { version = "0.27.0", features = ["serde"] }
```
Comment (lquerel) [inline Cargo.toml:79]: I used Tantivy for the search

**Comment by jsuereth** [comment]
@lquerel This is ready for review.  See my comments to your initial comments.  The goal of this is NOT completeness.  It's meant to be "bare bones" enough someone can take this and make it whatever in the future, but also NOT a burden for us to evolve weaver going forward if it exists.

This is why I didn't opt for anything more than iterator-based searching and only support displaying Attributes for now.  We should still have room to alter data model without updating rendering code here, which I expect we'll need as we worked towards "application schema" stabilziation.

File: src/registry/search.rs [hunk 3]
```diff
@@ -17,7 +44,358 @@ pub struct RegistrySearchArgs {
...
+        let title_block = Block::default()
+            .borders(Borders::TOP)
+            .style(Style::default().bg(Color::Black))
+            .border_style(Style::default().fg(Color::Gray))
+            .title_alignment(ratatui::layout::Alignment::Center)
+            .title_style(Style::default().fg(Color::Green))
+            .title("Weaver Search");
+        let title_contents = Line::from(vec![Span::styled(
+            format!(
+                "Loaded {0:?} registries w/ {1} attributes",
```
Comment (lquerel) [inline src/registry/search.rs:111]: Is the registry the entire set of groups starting with registry. or is each group starting with registry. considered an individual registry? If as I think it's the former then message should be something like "Loaded {0:?} groups w/ {1} attributes".

**Review by lquerel — APPROVED** [review]
See my comment regarding one of the messages displayed to the console.

**Pull request merged**